ReactWindows/ReactNative.Tests/ReactInstanceManagerTests.cs

```diff
@@ -232,6 +232,9 @@ public void ReactInstanceManager_Builder_SetterChecks()
 
             await DispatcherHelpers.CallOnDispatcherAsync(manager.DisposeAsync);
 
+            // The manager.DisposeAsync resets DispatherHelpers as well 
+            ReactNative.Bridge.DispatcherHelpers.Initialize();
+
```

Comment:
We have a very set of initialization steps. My hope was that we could encapsulate all the initialization of both the dispatcher and the choreographer in the app lifecycle events. It sounds like this is not the case.

If its not the case, then should we remove the initialization from the `OnResume` lifecycle event, and instead force people to initialize specifically (or do initialization in the ReactInstanceManager constructor to prevent breaking changes)?

Reply:
I tried to go with the minimum changes in the code AND to adapt the tests..
The app flow is complete: OnResume (called on creating or on resuming) initializes the helpers/choreographer, sometimes redundantly (on the next resume, etc.), Dispose uninitialize them (never really called for UWP).
Now, on top of that we have some random choreographer inits in the "CreateContext" methods, and I assumed they were due to having to call those methods from tests. They are NOT needed during normal app run.
Tests though call these objects without going through all the necessary steps, sometime the tests cover these very flows, so it's tricky.
Sometimes the test does something that I don't get (for ex. the test this comment as anchored on disposes the instance manager, THEN creates a context on it. Why? This is not normal flow!)

A more correct way would be:
- An Initialize added to instance manager, that inits helpers and choreographer. Ideally should be called once, so the "right" place is the Application constructor.. BUT the moment is extremely bad, you can't execute ApplicationView.MainView at that point, you get an exception. So app will have to call it at Create or Resume time anyway, maybe with some care to call it just once for the app lifecycle
- Dispose will continue to be paired with this.
- Resume/Suspend/etc. won't init/uninit anything like that anymore
- I'll remove various inits from "createContext" implementation and put the burden on tests to bring the lab mouse to expected start state.

Sounds ok?


---
In reply to: [170013099](https://github.com/Microsoft/react-native-windows/pull/1653#discussion_r170013099) [](ancestors = 170013099)